src/Orleans.Core/Async/AsyncExecutorWithRetries.cs

```diff
@@ -9,7 +9,7 @@ namespace Orleans
     /// This class is a convenient utility class to execute a certain asynchronous function with retries,
     /// allowing to specify custom retry filters and policies.
     /// </summary>
-    internal static class AsyncExecutorWithRetries
+    public static class AsyncExecutorWithRetries
```

Comment:
I'm not huge on exposing these since they are general utility classes. I'd be fine to copy them to the other places that they're needed or even include them via shared source.

Ultimately this is also ok if we don't like the alternatives.